devicetypes/zooz/zooz-zse11-q-sensor.src/zooz-zse11-q-sensor.groovy

```diff
@@ -0,0 +1,448 @@
+	if (cmd) {
+		zwaveEvent(cmd)
+	} else {
+		log.warn "Unable to parse: $description"
+	}
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCmd = cmd.encapsulatedCommand(commandClassVersions)
+	if (encapsulatedCmd) {
+		zwaveEvent(encapsulatedCmd)
+	} else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+	}
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpNotification cmd) {
+	log.debug "Device Woke Up"
+	List<String> cmds = []
+
+	if (state.pendingRefresh) {
+		state.pendingRefresh = false
+		cmds += getRefreshCmds()
+	}
+
+	cmds += getConfigCmds()
+	
+	if (!cmds) {
+		cmds << batteryGetCmd()
+	}
+
+	cmds << secureCmd(zwave.wakeUpV1.wakeUpNoMoreInformation())
+	sendHubCommand(cmds)
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	int val = (cmd.batteryLevel == 0xFF ? 1 : cmd.batteryLevel)
+	if (val > 100) {
+		val = 100
+	} else if (val < 1) {
+		val = 1
+	}
+
+	if (device.currentValue("powerSource") != "battery") {
+		log.debug "powerSource is battery"
+		sendEvent(name:"powerSource", value:"battery")
+	}
+
+	log.debug "battery is ${val}%"
+	sendEvent(name:"battery", value:val, unit:"%", isStateChange:true)
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	switch (cmd.sensorType) {
+		case tempSensorType:
+			def temp = convertTemperatureIfNeeded(cmd.scaledSensorValue, (cmd.scale ? "F" : "C"), cmd.precision)
+			sendTempEvent(temp)
+			break
+		case lightSensorType:
+			sendLightEvent(cmd.scaledSensorValue)
+			break
+		case humiditySensorType:
+			sendHumidityEvent(cmd.scaledSensorValue)
+			break
+		default:
+			log.debug "Unhandled: ${cmd}"
+	}
+}
+
+void sendTempEvent(reportedVal, boolean onlyIfNew=false) {
+	reportedVal = safeToDec(reportedVal)
+	state.reportedTemp = reportedVal
+
+	def adjVal = (safeToDec(settings?.tempOffset) + reportedVal)
+	if (!onlyIfNew || (adjVal != device.currentValue("temperature"))) {
+		log.debug "temperature is ${adjVal}°${temperatureScale}"
+		sendEvent(name:"temperature", value:adjVal, unit:temperatureScale)
+	}
+}
+
+void sendLightEvent(reportedVal, boolean onlyIfNew=false) {
+	reportedVal = safeToInt(reportedVal)
+	if (reportedVal < 0) {
+		// workaround for bug in original firmware
+		reportedVal = (reportedVal + 65536)
+	}
+	state.reportedLight = reportedVal
+
+	def adjVal = (safeToInt(settings?.lightOffset) + reportedVal)
+	if (adjVal < 0) adjVal = 0
+	if (!onlyIfNew || (adjVal != device.currentValue("illuminance"))) {
+		log.debug "illuminance is ${adjVal}lux"
+		sendEvent(name:"illuminance", value:adjVal, unit:"lux")
+	}
+}
+
+void sendHumidityEvent(reportedVal, boolean onlyIfNew=false) {
+	reportedVal = safeToInt(reportedVal)
+	state.reportedHumidity = reportedVal
+
+	def adjVal = (safeToInt(settings?.humidityOffset) + reportedVal)
+	if (adjVal < 0) adjVal = 0
+	if (adjVal > 100) adjVal = 100
```

Comment:
what about ` adjVal = Math.min(Math.max(0, adjVal ), 100)` ?

Reply:
I've made the requested change